blade-egui/src/lib.rs

```diff
@@ -128,7 +129,11 @@ impl GuiPainter {
     /// It supports renderpasses with only a color attachment,
     /// and this attachment format must be The `output_format`.
     #[profiling::function]
-    pub fn new(info: blade_graphics::SurfaceInfo, context: &blade_graphics::Context) -> Self {
+    pub fn new(
+        info: blade_graphics::SurfaceInfo,
+        context: &blade_graphics::Context,
+        sample_count: u32,
```

Comment:
I don't think this is useful. Egui is used as HUD, and this is practically never rendered in MSAA-enabled surface. MSAA is needed to fight the aliasing of geometry and specularity - neither are a problem for UI in HUD. So the proper thing to do is to always resolve your main renderer, and then draw HUD on top (without multi-sampling).

Reply:
Please explain a little more. I thought the geometry produced by egui still has edges that could potentially alias. Even if less noticable since the lines are always straight to my knowledge and thus bad cases are mostly avoided, I don't know if rounded corners and stuff are done through geometry or bitmaps.

The docs of eframe mention that one might want to turn on anti-aliasing even if the textures and fonts within egui already are anti-aliased

https://docs.rs/eframe/latest/eframe/struct.NativeOptions.html#structfield.multisampling

Reply:
I don't think straight horizontal/vertical lines can be aliased. You'd only see the aliasing issue if the stuff is moving, or it's not already aligned to the pixel grid. The advice in their docs is about displaying 3D content within egui, which is arguably a very niche case. Unless your application needs it, let's consider this out of scope and not bother changing blade-egui.

Reply:
Yeah while moving, like dragging a window is mostly the case I thought about.

3D embedding would be cool but not something I am doing currently xD

Reply:
If egui is used for the actual game UI, then it's unlikely there is going to be any resizing at all.
If it's used as debug UI. then aliasing of movement doesn't matter, since it's not shipping.
Neither case is also fine with the MSAA overhead for these rectangles.
Smooth edges can be achieved by much simpler means, like analytical anti-aliasing. E.g. we know that this line is barely touching the pixel center so we'll make it more transparent. With this, you get smooth movement without any overhead.

Reply:
We are using egui for our entire engine:s editor and have lots of movable windows and whatnot, part of the editor is also intended to be shipped.

Whether masaa for egui is the best and most efficient solution or not is separate :3